### ISSUE TITLE
Detect string formatting with number

### ISSUE THREAD

**Issue by unformatt** [gforcada/flake8-pep3101#27]
Is there a rule I need to set?

```bash
echo "print 'asd %s' % 1\nfoo()" > test.py
flake8 test.py
test.py:2:1: F821 undefined name 'foo'
```

Versions:
```
flake8 --version
3.6.0 (flake8_pep3101: 1.2.1, flake8_quotes: 2.1.1, mccabe: 0.6.1, pycodestyle: 2.4.0, pyflakes: 2.0.0) CPython 2.7.15 on Darwin```

**Comment by gforcada** [comment]
Hi! thanks for reporting the issue!

I'm a bit busy right now, I will try to have an answer within a few days, give me a friendly ping if you haven't heard about me after that timeframe.

**Comment by unformatt** [comment]
Hi @gforcada were you able to see the same problem? Let me know if there's any debugging I can do for you.

**Comment by gforcada** [comment]
Sorry for the long delay, puzzlingly enough, I can reproduce it as well locally, I will try to have a look right now :slightly_smiling_face:

**Comment by gforcada** [comment]
@unformatt now that I had a closer look, it is crystal clear :slightly_smiling_face: the *problem* of your code is that you are using a number, and at least right now, to avoid false positives, this flake8 plugin skips any modulo statement that has numbers on it, so all the usual `index % 50 == 0` would be always skip.

We could add another case to report whenever the left hand operator of a modulo is a string and the right hand is a number.

We will still have problems if you do:
```
text = 'my long text that needs a number right here %i for formatting purposes'
print(text % 55)
```
This above will be quite hard to get, as we would have to get the text variable and check what it actually holds to know if it should warn or not :confused: 

At least, though, a fix for your case should not be too much work :tm:

**Comment by gforcada** [comment]
Released on `1.3.0` :tada:

**Comment by unformatt** [comment]
Thanks @gforcada !
